Subject filter UI improvements: LFS-756, LFS-764

veronikaslc opened the pull request:
[LFS-756](https://phenotips.atlassian.net/browse/LFS-756) SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog
[LFS-764](https://phenotips.atlassian.net/browse/LFS-764) Once a subject is selected in the Subjects filter, its full id should be displayed both in the filter input and the filter chip after the dialog is closed

Commit:
LFS-756: SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog

marta- reviewed (changes requested):
The Icon has changed and looks good, but the full id is not displayed and "See all results" is still there.

![image](https://user-images.githubusercontent.com/651980/101191671-f7908480-3627-11eb-9935-880482b78cc0.png)

In the screenshot, tumors labeled "1" from 3 different patients come up as results.

Commit:
LFS-756: SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog
Refactored to avoid code duplication and reuse existing component

marta- commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
```diff
@@ -259,10 +261,10 @@ function SearchBar(props) {
                     </MenuItem>
                   ))}
-                  { !results[0]?.disabled &&
+                  { !results[0]?.disabled && showAllResultsLink &&
```
This is off topic, but it looks like our administration setting for displaying total number of results is completely ignored.

![image](https://user-images.githubusercontent.com/651980/101232261-a95aa000-367e-11eb-85fd-cc3c01e07763.png)

veronikaslc replied:
What is the expected behaviour that does not work?

sashaandjic replied:
> What is the expected behaviour that does not work?

New issue logged in #https://phenotips.atlassian.net/browse/LFS-766

marta- replied:
The expected behavior was for the link at the bottom of the quick search dropdown to read "See all results" instead of "N more results". I see now that it was a caching issue, i.e. the app needs to be refreshed for the setting to be taken into account, which is fine since it's not a setting that will keep changing and needs to be immediately acknowledged.

marta- reviewed (commented):
This looks right and works per specs.

There's one minor alignment issue - the icon is too low and should be aligned with the text:
![image](https://user-images.githubusercontent.com/651980/101232520-0acf3e80-3680-11eb-9ce2-fe15bc2dddb8.png)


One more issue has been noticed since reporting this one: LFS-724 - Uses should be able to select the item from the dropdown instead of having to type the whole thing in the input.

Assuming this would affect whether `disableLink` stays as a parameter the way it is implemented in this PR or gets replaced by another parameter (e.g. `onItemClick`), could LFS-764 be fixed here too?

veronikaslc commented:
I think LFS-764 is fixed here too.

Commit:
LFS-756: SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog
Allign dropdown text to the middle

sashaandjic commented:
> I think LFS-764 is fixed here too.

It doesn't seem to work. When I click on any entry in the drop down list, nothing happens. LFS-756 issue is resolved.
![image](https://user-images.githubusercontent.com/23243244/101270775-67972b80-374a-11eb-9e57-13f38fe9cbb4.png)

Commit:
LFS-764: Modify filters dialog improvement

sashaandjic commented:
The short ID (screenshot 2 & 3) needs to be replaced with the long one (screenshot 1):
![image](https://user-images.githubusercontent.com/23243244/101375758-c610f680-387d-11eb-875f-0afdf9a68563.png)
![image](https://user-images.githubusercontent.com/23243244/101375785-cf9a5e80-387d-11eb-802e-d59efd15e9e5.png)
![image](https://user-images.githubusercontent.com/23243244/101375812-d4f7a900-387d-11eb-90df-a7eb032ebf1a.png)

Commit:
LFS-756: SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog
Set full entity id value to selected id

sashaandjic reviewed (approved):
tested

sdumitriu commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
```diff
@@ -42,15 +43,17 @@ const LFS_QUERY_TEXT_KEY = "text";
  * @param {func} onSelectFinish Function to call after onSelect
  * @param {func} queryConstructor Function that takes (query, requestID) and returns a URL to query for suggestions. Default: use /query?query
  * @param {func} resultConstructor Function that constructs a DOM element from a row of results.
+ * @param {bool} showAllResultsLink If true, show the link “See all results” of the bottom o the results dropdown
```
```suggestion
 * @param {bool} showAllResultsLink If true, show the link “See all results” of the bottom of the results dropdown
```

sdumitriu commented on modules/commons/src/main/frontend/src/headerStyle.jsx:
```diff
@@ -97,6 +97,13 @@ const headerStyle = theme => ({
   dropdownItem: {
     whiteSpace: "normal"
   },
+  dropdownItemMiddleAligned: {
```
These class names are very specific...

sdumitriu commented on modules/commons/src/main/frontend/src/headerStyle.jsx:
```diff
@@ -97,6 +97,13 @@ const headerStyle = theme => ({
+    padding: 0
+  },
+  dropdownItemTextMiddleAligned: {
+    marginTop: "14px"
```
Can't this be done in theme spacing units?

veronikaslc committed:
LFS-756 SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog

Typo fix

Co-authored-by: Sergiu Dumitriu <sdumitriu@users.noreply.github.com>

marta- committed:
LFS-756: SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog
Removed unnecessary CSS classes

marta- committed:
LFS-756: SubjectID and corresponding icon are incorrect in the Subject dropdown of the Modfiy filters dialog
Better padding distribution in nested dropdown items

sashaandjic reviewed (approved):
tested

marta- committed:
LFS-764 Once a subject is selected in the Subjects filter, its full id should be displayed both in the filter input and the filter chip after the dialog is closed
Removed side effect: Quick search input should not be populated with the selected item value, only selectors should have this behavior.

sashaandjic reviewed (approved):
tested

sashaandjic reviewed (commented):
tested
